components/base/wheeled/kinematics.go

```diff
@@ -51,24 +68,82 @@ func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
 }
 
 func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
-	// TODO(RSDK-2311): complete the implementation
-	return []referenceframe.Input{}, errors.New("not implemented yet")
+	// TODO(rb): make a transformation from the component reference to the base frame
+	pose, _, err := kwb.slam.GetPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pt := pose.Point()
+	theta := math.Mod(pose.Orientation().OrientationVectorRadians().Theta, 2*math.Pi) - math.Pi
+	return []referenceframe.Input{{Value: pt.X}, {Value: pt.Y}, {Value: theta}}, nil
 }
 
-func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	// TODO(RSDK-2311): complete the implementation
-	return errors.New("not implemented yet")
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, desired []referenceframe.Input) (err error) {
+	// this loop polls the error state and issues a corresponding command to move the base to the objective
+	// when the base is within the positional threshold of the goal, exit the loop
+	// TODO(rb): check for the context being cancelled and stop if so
+	for {
+		current, err := kwb.CurrentInputs(ctx)
+		if err != nil {
+			return err
+		}
+
+		// get to the x, y location first
+		desiredHeading := math.Atan2(current[1].Value-desired[1].Value, current[0].Value-desired[0].Value)
```

Comment:
Since we're using Spin and MoveStraight, do we need to account for the base thinking Y is forwards? Or is that not until SetVelocity comes into play?

Either way can we add a comment here talking about the Y-forwards-ness of the base?

Reply:
Yeah I'll add a comment.  It doesn't really matter now what it thinks is forward since we're using MoveStraight and Spin